trenches2.py

```diff
@@ -565,9 +565,231 @@ def __init__(self, trench_corners: Dict[int, TrenchCorner], trenches: List[Trenc
+                street_trenches[street_name][i] = trench
+
+    # Loop over every building and try and find the road trench that is closest to it
+    # and find the intersection point between the road trench and a perpendicular line
+    # of the road trench that goes through the building centroid
+    building_by_closest_trench: Dict[int, List[Trench_info]] = dict()
+    for building_index, building in building_gdf.iterrows():
+        closest_trench_info = None
+        street_name = building['addr:street']
+        centroid = building['geometry'].centroid
+        distance = float('inf')
+        building_centriod_node = {'x': centroid.xy[0][0], 'y': centroid.xy[1][0], 'building_index': building_index}
+        # There might be buildings in the box that are on roads that are not in the box, geo fencing problem
+        if len(street_trenches[street_name]) > 0:
+            # Loop over every trench for this street and find the closest one
+            for trench_index, trench in street_trenches[street_name].items():
+                if trench_index not in building_by_closest_trench:
+                    building_by_closest_trench[trench_index] = list()
+                corner_u: TrenchCorner = corner_by_id[trench['u_for_edge']]
+                corner_v: TrenchCorner = corner_by_id[trench['v_for_edge']]
+                if 'geometry' not in trenches[trench_index]:
+                    # Get the intersection point between the road trench and a perpendicular line of the building
+                    perpendicular_line = get_perpendicular_line(corner_u, corner_v, building_centriod_node)
+                    projected = get_intersection_point2(perpendicular_line, (corner_u, corner_v))
+                    # Extra check to make sure we are not doing something wrong, might be a bug in the code
+                    if is_between2(corner_u, corner_v, projected):
+                        new_distance = node_distance(projected, building_centriod_node)
+                        # Check if this trench is the closest one so far
+                        if new_distance < distance:
+                            new_v_node = projected
+                            distance = new_distance
+                            closest_trench = trench_index
+                            closest_trench_info = {'building_centroid_node': building_centriod_node,
+                                                   'ref_new_v_node': new_v_node,
+                                                   'closest_trench': closest_trench,
+                                                   'geometry': False,
+                                                   'ref_corner_u': corner_u,
+                                                   'segment_index': None}
+
+                else:
+                    # This is an attempt of finding a closest trench but for trenches that have geometry
+                    # it is not used if the g_box is no simplified
```

Comment:
Does this mean this part of the code is redundant now that we use simplify=True?

Reply:
Well I wanted to keep it in since we might need the Simplify=True if the network is as big as a city